### PR TITLE
update profile

### PR DESCRIPTION
**Pull request by Dawaic6**
#### Implemented Features:
- User Profile Update API Endpoint 
- Unit test  for update  Profile
- swagger document  for  Update profile  endpoint

#### Definition of Done:
- User Profile Update API Endpoint (where you can update firstName, lastName , email  by finding user by id )
- Unit test  
- swagger document  for  Update profile  endpoint

Related Issues:
#28

### To use the Code ###
- **Clone this repository**:
  git clone https://github.com/atlp-rwanda/dynamites-ecomm-be.git
- **Run**:
`npm run dev`
- **Test-covarage**:
`npm run test:ci`
- **Test**:
`npm run test`
### Screenshot
-**for test**
image

-**for swagger**
image

### CODE REVIEW

**Review by niyobern — CHANGES_REQUESTED** [review]
change path to be sub route of user instead of looking like a a separate entity.
You may need to change it to something like like `api/v1/user/{userId}/profile` or `api/v1/user/{userId}`

File: src/docs/userRegisterDocs.ts [hunk 1]
```diff
@@ -170,3 +170,77 @@
  *         description: An error occurred while deleting the record.
  */
 
+
+/**
+ * @swagger
+ * /api/v1/updateProfile/{id}:
```
Comment (wayneleon1) [inline src/docs/userRegisterDocs.ts:227]: since it coming from userRoutes you have to write this endpoint like this `/api/v1/user/updateProfile/{id}`

**Review by wayneleon1 — CHANGES_REQUESTED** [review]
make changes at  src/docs/userRegisterDocs.ts  on line 176

File: src/docs/userRegisterDocs.ts (hunk 1 above)
Comment (Dawaic6) [reply src/docs/userRegisterDocs.ts:227]: yes, finished

**Comment by Dawaic6** [comment]
> change path to be sub route of user instead of looking like a a separate entity. You may need to change it to something like like `api/v1/user/{userId}/profile` or `api/v1/user/{userId}`

done

**Review by niyobern — DISMISSED** [review]

**Review by ambroisegithub — DISMISSED** [review]
It Looks Cool For to be merged..

**Review by 13XAVI — DISMISSED** [review]
Good

**Review by jkarenzi — APPROVED** [review]

**Review by ambroisegithub — APPROVED** [review]
It Looks good for merge!!!

**Review by dushimeemma — APPROVED** [review]
Good job 🥇

**Review by wayneleon1 — APPROVED** [review]
great job

**Pull request merged**